examples/main/main.cpp

```diff
@@ -241,10 +246,7 @@ int main(int argc, char ** argv) {
         sigint_action.sa_flags = 0;
         sigaction(SIGINT, &sigint_action, NULL);
 #elif defined (_WIN32)
-        auto console_ctrl_handler = [](DWORD ctrl_type) -> BOOL {
```

Comment:
try to add a `+` before the lambda
`+[](DWORD ctrl_type) -> BOOL {`

Reply:
this ensures it is stateless :)

Reply:
Thanks! I do prefer the lambda as it keeps the _WIN32 code in one block instead of all over the code. Let me change this real quick and have him try it out.

Reply:
Man, just when I thought I knew C++